.github/workflows/changelog.yaml

```diff
@@ -0,0 +1,43 @@
+name: changelog entry
+
+on:
+  pull_request:
+    types: [opened, synchronize, labeled, unlabeled, reopened]
+    branches:
+      - master
+
+permissions:
+  contents: read
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check-changelog-entry:
+    name: Check for changelog entry
+    runs-on: ubuntu-latest
+    if: contains(github.event.pull_request.labels.*.name, 'skip news') != true
+
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # towncrier needs a non-shallow clone
+          fetch-depth: '0'
+
+      - name: Set up environment
+        uses: ./.github/actions/setup-env
+        with:
+          python-version: '3.9'
+          nox-setup: "tools"
+
+      - name: Check for presence of a Change Log fragment (only pull requests)
+        # NOTE: The pull request' base branch needs to be fetched so towncrier
+        # is able to compare the current branch with the base branch.
+        run: |
+          if ! towncrier check --compare-with origin/${BASE_BRANCH}; then
```

Comment:
Hmm, we could combine this with the original check, as towncrier does not check whether the PR number matches

Reply:
I don't think checking that number is entirely worth it, can be a manual check for now as we prefer using an issue number if one exists.